daemons/gptp/common/avbts_message.hpp

```diff
@@ -223,21 +226,50 @@ class PTPMessageCommon {
 
 #pragma pack(push,1)
 
+#define PATH_TRACE_TLV_TYPE 0x8
+
 class PathTraceTLV {
  private:
 	uint16_t tlvType;
-	uint16_t lengthField;
-	ClockIdentity identity;
+	typedef std::list<ClockIdentity> IdentityList;
+	IdentityList identityList;
  public:
-	 PathTraceTLV() {
-		tlvType = PLAT_htons(0x8);
-		lengthField = PLAT_htons(sizeof(identity));
-	}
-	void setClockIdentity(ClockIdentity * id) {
-		identity = *id;
+	PathTraceTLV() {
+		tlvType = PLAT_htons(PATH_TRACE_TLV_TYPE);
+	}
+	void parseClockIdentity(uint8_t *buffer) {
+		int length = PLAT_ntohs(*((uint16_t *)buffer))/PTP_CLOCK_IDENTITY_LENGTH;
+		buffer += sizeof(uint16_t);
+		for(; length > 0; --length) {
+			ClockIdentity add;
+			add.set(buffer);
+			identityList.push_back(add);
+			buffer += PTP_CLOCK_IDENTITY_LENGTH;
+		}
+	}
+	void appendClockIdentity(ClockIdentity * id) {
+		identityList.push_back(*id);
 	}
 	void toByteString(uint8_t * byte_str) {
-		memcpy(byte_str, this, sizeof(*this));
+		IdentityList::iterator iter;
+		*((uint16_t *)byte_str) = tlvType;
```

Comment:
Add a comment /* tlvType already in network byte order */. But that does bring up question for future documentation and/or resolution. What is the gPTP policy for storing variables? Are some random variables stored in network order? Code I have worked with stores all variables in machine order until such time as a packet is constructed. Only the packet under construction contains network ordered data.